SourceCode/GPS/Classes/CABCurve.cs

```diff
@@ -177,36 +181,44 @@ public async void BuildCurveCurrentList(vec3 pivot)
             {
-                if (!isBusyWorking)
-                {
-                    //is boundary curve - use task
-                    isBusyWorking = true;
-                    isCurveValid = true;
-                    lastHowManyPathsAway = howManyPathsAway;
-                    lastIsHeadingSameWay = isHeadingSameWay;
-                    double distAway = widthMinusOverlap * howManyPathsAway + (isHeadingSameWay ? -mf.tool.offset : mf.tool.offset) + track.nudgeDistance;
+                //is boundary curve - use task
+                isCurveValid = true;
+                lastHowManyPathsAway = howManyPathsAway;
+                lastIsHeadingSameWay = isHeadingSameWay;
+                double distAway = widthMinusOverlap * howManyPathsAway + (isHeadingSameWay ? -mf.tool.offset : mf.tool.offset) + track.nudgeDistance;
 
-                    distAway += (0.5 * widthMinusOverlap);
+                distAway += (0.5 * widthMinusOverlap);
 
-                    curList = await Task.Run(() => BuildNewOffsetList(distAway, track));
-                    isBusyWorking = false;
-                    findGlobalNearestCurvePoint = true;
+                if (cts != null)
+                {
+                    cts.Cancel();
+                }
+                cts = new CancellationTokenSource();
 
-                    if (mf.isSideGuideLines && mf.camera.camSetDistance > mf.tool.width * -400)
-                    {
-                        //build the list list of guide lines
-                        guideArr = await Task.Run(() => BuildCurveGuidelines(distAway, mf.ABLine.numGuideLines, track));
-                    }
-                    else
-                    {
-                        guideArr?.Clear();
-                    }
+                if (build != null) await build;
+
+                build = Task.Run(() => BuildNewOffsetList(distAway, track, cts.Token), cts.Token);
+                curList = await build;
+                findGlobalNearestCurvePoint = true;
 
+                if (mf.isSideGuideLines && mf.camera.camSetDistance > mf.tool.width * -400)
+                {
+                    if (buildList != null)
+                        await buildList;
+                    //build the list list of guide lines
+                    buildList = Task.Run(() => BuildCurveGuidelines(distAway, mf.ABLine.numGuideLines, track, cts.Token), cts.Token);
+                    guideArr = await buildList;
+                }
+                else
+                {
+                    if (buildList != null) await buildList;
+                    guideArr?.Clear();
                 }
             }
         }
 
-        public List<vec3> BuildNewOffsetList(double distAway, CTrk track)
+        
+        public List<vec3> BuildNewOffsetList(double distAway, CTrk track, CancellationToken ct = new CancellationToken())
```

Comment:
Usually `CancellationToken.None` is used if the token is optional.

```suggestion
        public List<vec3> BuildNewOffsetList(double distAway, CTrk track, CancellationToken ct = CancellationToken.None)
```

Reply:
Oh, I'm sorry I did not remember that correctly (and I should have double-checked).

Maybe use `default` instead?
```suggestion
        public List<vec3> BuildNewOffsetList(double distAway, CTrk track, CancellationToken ct = default)
```